pytorch_lightning/profiler/pytorch.py

```diff
@@ -522,7 +509,7 @@ def _delete_profilers(self) -> None:
     def teardown(self, stage: Optional[str] = None) -> None:
         self._delete_profilers()
 
-        for k in self._recording_map:
+        for k in list(self._recording_map.keys()):
```

Comment:
```suggestion
        for k in list(self._recording_map):
```

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/11268#discussion_r788540655

Reply:
You are right. :)